parquet/src/file/reader.rs

```diff
@@ -86,6 +95,14 @@ pub trait RowGroupReader: Send + Sync {
     /// Get page reader for the `i`th column chunk.
     fn get_column_page_reader(&self, i: usize) -> Result<Box<dyn PageReader>>;
 
+    /// Get page reader for the `i`th column chunk with offset index(in one row group).
+    /// `row_group_pages_offset_index` construct from pageIndex and input rowRanges, here we skip needless page.
+    fn get_column_page_reader_with_offset_index(
+        &self,
+        column_index: usize,
+        row_group_pages_offset_index: &[FilterOffsetIndex],
```

Comment:
```suggestion
      page_offset_index : &FilterOffsetIndex,
```
a reference of struct is enough.
I can't figure out a good name for parameter.